rectools/models/base.py

```diff
@@ -430,12 +469,24 @@ def recommend(
             If some of given users are warm/cold and model doesn't support such type of users and
             `on_unsupported_targets` is set to "raise".
         """
+        if self.require_recommend_context and (context is None):
+            raise ValueError(
+                f"Mismatch between require_recommend_context ({self.require_recommend_context}) "
+                f"and context presence ({context is not None}). "
+                "Context must be provided when 'require_recommend_context' is True."
+            )
+        if not self.require_recommend_context and (context is not None):
+            warnings.warn(
+                "You are providing context to a model that does not require it."
+                " Some models may replace it with None ",
```

Comment:
```suggestion
                "Context is set to ``None``",
```
Please set context to None here, only once in the base models so that we don't have to worry about it anymore in any of the inherited models.